### ISSUE TITLE
Automatically determine :ssl based on connection uri

### ISSUE THREAD

**Issue by reify-james-vickers** [protojure/lib#133]
`protojure.grpc.client.providers.http2/connect` takes a map that must contain `:uri` and can optionally contain `:ssl` with boolean value.  It seems possible to automatically know that SSL should be used if the uri starts with `https://` which is defining the connection scheme as HTTPS/SSL.  Would it be possible to not have to provide `:ssl` in such cases then?

**Comment by ghaskins** [comment]
Yes, definitely possible.  The ssl field should remain for backwards compat, but we could make it have a default based on the scheme.  I would consider a PR that implements this.

**Comment by reify-james-vickers** [comment]
https://github.com/protojure/lib/pull/136
